flux/src/flagger/canarystatus.tsx

```diff
@@ -0,0 +1,22 @@
+import React from 'react';
+import { StatusLabel } from '@kinvolk/headlamp-plugin/lib/components/common';
+
+export default function CanaryStatus({ status }) {
+    let type;
+    switch (status) {
+        case 'Progressing':
+            type = 'warning';
+            break;
+        case 'Succeeded':
+            type = 'success';
+            break;
+        case 'Failed':
+            type = 'error';
+            break;
+        default:
+            type = '';
```

Comment:
Nit: Normally I use a const object for this, corresponding { Progressing: 'warning' , ... } for example. The switch is okay though.